File: src/p1221/python/solution.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        result, weight = 0, 0
        for x in s:
            if x == 'L':
                weight -= 1
            else:
                weight += 1
            if weight == 0:
                result += 1
        return result

File: src/p1221/golang/solution.go
func balancedStringSplit(s string) int {
	var result, weight int
	for i := 0; i < len(s); i++ {
		if s[i] == 'L' {
			weight--
		} else {
			weight++
		}
		if weight == 0 {
			result++
		}
	}
	return result
}
